logstash-core/lib/logstash/logging/logger.rb

```diff
@@ -79,5 +80,35 @@ def self.get_logging_context
         return @@logging_context
       end
     end
+
+    class SlowLogger
+      def initialize(name, warn_threshold, info_threshold, debug_threshold, trace_threshold)
+        slowlog_name = ["slowlog", name].join('.')
+        @slowlogger = LogManager.getLogger(slowlog_name)
+        @warn_threshold = warn_threshold
+        @info_threshold = info_threshold
+        @debug_threshold = debug_threshold
+        @trace_threshold = trace_threshold
+      end
+
+      def as_data(event, time_in_nanos)
```

Comment:
right, this is a general problem of context that we have in the pipeline, many of the changes coming in the LIR fix this.

But, this brings up my question above: https://github.com/elastic/logstash/pull/6128#issuecomment-256767708.
would it make more sense to have a slowlog capture the whole `filter_func` instead of watching time on a per plugin basis?

For now, I will add the ID, and took_in_millis.